policies/templates/gcp_allowed_resource_types_v1.yaml

```diff
@@ -0,0 +1,122 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This template is for policies restricting the resource types
+# in your organization hierarchy. You can specify which resource
+# types are allowed (mode: whitelist) or denied (mode: blacklist)
+# using the "resource_types_list" and "mode" parameters.
+
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: gcp-allowed-resource-types-v1
+spec:
+  crd:
+    spec:
+      names:
+        kind: GCPAllowedResourceTypesConstraintV1
+        plural: gcpallowedresourcetypesconstraintsv1
+      validation:
+        openAPIV3Schema:
+          properties:
+            mode:
+              type: string
+              enum: [whitelist, blacklist]
+              description: "String identifying the operational mode,
+              whitelist or blacklist. In the whitelist mode, only resource types
+              from the resource_types_list will be allowed (all other types
+              will raise a violation). In the blacklist mode, any resource
+              type not in the resource_types_list will not raise a violation."
+            resource_types_list:
```

Comment:
maybe `resource_type_list` is better?

Reply:
renamed to resource_type_list. Thanks, I was not sure about it